homeassistant/components/zimi/config_flow.py

```diff
@@ -0,0 +1,112 @@
+"""Config flow for zcc integration."""
+
+from __future__ import annotations
+
+import logging
+import socket
+from typing import Any
+
+import voluptuous as vol
+from zcc import ControlPointDiscoveryService, ControlPointError
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_PORT
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.device_registry import format_mac
+
+from . import async_connect_to_controller
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_HOST): str,
+        vol.Optional(CONF_PORT, default=5003): int,
+        vol.Required(CONF_MAC): str,
+    }
+)
+
+
+class ZimiConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for zcc."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> config_entries.ConfigFlowResult:
+        """Handle the initial step."""
+
+        errors: dict[str, str] = {}
+        description_placeholders: dict[str, str] = {}
+
+        if user_input is not None:
+            data: dict[str, Any] = {"title": "ZIMI Controller"}
+
+            try:
+                if not user_input[CONF_HOST]:
+                    try:
+                        description = await ControlPointDiscoveryService().discover()
```

Comment:
how does discovery work here? Can't we use one of HA's built-in discovery methods?

Reply:
Discovery uses the native ZIMI process with UDP broadcast to a known port and UDP unicast back with zcc details.

It is unique to zimi.